data-service/src/main/java/com/amazonaws/blox/dataservice/handler/EnvironmentHandler.java

```diff
@@ -30,26 +31,71 @@
 
   @NonNull private EnvironmentRepository environmentRepository;
 
-  public Environment createEnvironment(final String environmentName, final String accountId)
-      throws EnvironmentExistsException, StorageException {
-    return null;
+  public Environment createEnvironment(final Environment environment)
+      throws EnvironmentExistsException, ServiceException {
+    try {
+      return environmentRepository.createEnvironment(environment);
+    } catch (final StorageException e) {
+      throw new ServiceException(
+          String.format(
+              "Exception occurred when creating environment with id %s and version %s",
+              environment.getEnvironmentId(), environment.getEnvironmentVersion()),
+          e);
+    }
   }
 
-  public Environment describeEnvironment(final String environmentId) {
-    return null;
+  public Environment createEnvironmentTargetVersion(final Environment environment)
+      throws EnvironmentExistsException, ServiceException {
+    try {
+      return environmentRepository.createEnvironmentTargetVersion(environment);
+    } catch (final StorageException e) {
+      throw new ServiceException(
+          String.format(
+              "Exception occurred when creating environment target version with id %s",
+              environment.getEnvironmentId()),
+          e);
+    }
+  }
+
+  public Environment describeEnvironmentTargetVersion(final String environmentId)
+      throws EnvironmentNotFoundException, ServiceException {
+    try {
+      return environmentRepository.getEnvironmentTargetVersion(environmentId);
+    } catch (final StorageException e) {
+      throw new ServiceException(
+          String.format("Exception occurred when getting environment with id %s", environmentId),
+          e);
+    }
   }
 
   public Environment describeEnvironment(
-      final String environmentId, final String environmentVersion) {
-    return null;
+      final String environmentId, final String environmentVersion)
+      throws EnvironmentNotFoundException, ServiceException {
+    try {
+      return environmentRepository.getEnvironment(environmentId, environmentVersion);
+    } catch (final StorageException e) {
+      throw new ServiceException(
+          String.format(
+              "Exception occurred when getting environment with id %s and version %s",
+              environmentId, environmentVersion),
+          e);
+    }
   }
 
-  public List<String> listClustersWithEnvironments() {
-    return null;
+  public List<String> listEnvironmentIdsWithCluster(final String cluster) throws ServiceException {
```

Comment:
Since most AWS API's that start with `list` return `Id`s of some sort, should this maybe just be `listEnvironmentsWithCluster`?

Reply:
sure